### PR TITLE
Dev lunatic reader update

### PR DESCRIPTION
**Pull request by loichenninger**

### CODE REVIEW

File: src/main/java/fr/insee/bpm/metadata/reader/lunatic/LunaticUtils.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+    public static String getFirstResponseName(JsonNode components) {
+        for (JsonNode component : components) {
+            ComponentLunatic componentLunatic = ComponentLunatic.fromJsonName(component.get(Constants.COMPONENT_TYPE).asText());
+            if (componentLunatic == ComponentLunatic.QUESTION) {
+                String result = getFirstResponseName(component.get(Constants.COMPONENTS));
+                if (result != null) return result;
+            } else {
+                if (component.has(RESPONSE)) {
+                    return component.get(RESPONSE).get("name").asText();
```
Comment (alicela) [inline src/main/java/fr/insee/bpm/metadata/reader/lunatic/LunaticUtils.java:40]: use getVariableName

File: src/main/java/fr/insee/bpm/metadata/reader/lunatic/LunaticReader.java [hunk 2]
```diff
@@ -149,7 +138,13 @@ public static MetadataModel getMetadataFromLunatic(InputStream lunaticFile) {
 			rootNode = JsonReader.read(lunaticFile);
 			List<String> variables = new ArrayList<>();
 			JsonNode variablesNode = rootNode.get(VARIABLES);
-			variablesNode.forEach(newVar -> variables.add(newVar.get("name").asText()));
+			variablesNode.forEach(newVar -> {
+				String varName = newVar.get("name").asText();
```
Comment (alicela) [inline src/main/java/fr/insee/bpm/metadata/reader/lunatic/LunaticReader.java:142]: use LunaticUtils.getVariableName

File: src/main/java/fr/insee/bpm/metadata/reader/lunatic/LunaticUtils.java [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+     * @param similarStrings : list of strings
+     * @return the common prefix
+     */
+    public static String findLongestCommonPrefix(List<String> similarStrings) {
+        int minLength = similarStrings.getFirst().length();
+        for(String str : similarStrings){
+            if (str.length()<minLength){
+                minLength = str.length();
+            }
+        }
```
Comment (alicela) [inline src/main/java/fr/insee/bpm/metadata/reader/lunatic/LunaticUtils.java:85]: int minLength = similarStrings.stream()
                              .mapToInt(String::length)
                              .min()
                              .orElse(similarStrings.getFirst().length());

File: src/main/java/fr/insee/bpm/metadata/reader/lunatic/LunaticUtils.java [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+        String commonPrefix="";
+        for(int i=1;i<minLength;i++){
+            boolean isCommon=true;
+            String stringToTest = similarStrings.getFirst().substring(0,i);
+            for (String str : similarStrings){
+                if (!str.startsWith(stringToTest)){
```
Comment (alicela) [inline src/main/java/fr/insee/bpm/metadata/reader/lunatic/LunaticUtils.java:91]: Replace for and if by lambda 

boolean isCommon = similarStrings.stream()
                                 .allMatch(str -> str.startsWith(stringToTest));

File: src/main/java/fr/insee/bpm/metadata/reader/lunatic/LunaticReader.java (hunk 2 above)
Comment (loichenninger) [reply src/main/java/fr/insee/bpm/metadata/reader/lunatic/LunaticReader.java:142]: It won't work here, the node of a variable does not have a RESPONSE attribute.

**Review by alicela — APPROVED** [review]

**Pull request merged**